airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_source_adapter.py

```diff
@@ -58,6 +59,6 @@ def _select_abstract_streams(self, config: Mapping[str, Any], configured_catalog
                     f"The stream {configured_stream.stream.name} no longer exists in the configuration. "
                     f"Refresh the schema in replication settings and remove this stream from future sync attempts."
                 )
-            if isinstance(stream_instance, StreamFacade):
+            if isinstance(stream_instance, (StreamFacade, FileBasedStreamFacade)):
```

Comment:
this makes me think we need an abstraction on top of the two facades with a single method to get a abstract stream. 

maybe something like
```
class AbstractStreamFacade[Generic[StreamType]:
   @abstractmethod    
   def get_underlying_stream -> StreamType:
        ...
```
where StreamType could either be StreamFacade or FileBasedStreamFacade

Reply:
On a sync with @clnoll, I also proposed that we don't use the facade pattern. We used it for the CCDK at first because it's too much work to migrate to the new model so we have a layer to translate the old code to the new. I was wondering if we could just migrate the FCDK to the `AbstractStream` interface and not use the facade

Reply:
@maxi297 I've gone in a different direction, which allows us to have one `DefaultStream` but still uses a `FileBasedStreamFacade`. I think this is more in line with the existing structure.